crates/nu-utils/standard_library/tests.nu

```diff
@@ -37,20 +29,55 @@ def main [
     --command: string, # Test command to run. Default: all test command found in the files.
     --list, # Do not run any tests, just list them (dry run)
 ] {
-    let dry_run = ($list | default false)
-    for test_file in (collect-modules $path $module) {
-        let $module_name = ($test_file | path parse).stem
+    let tests = (
+        ls ($path | default $env.FILE_PWD | path join "test_*.nu")
+        | each {|row| {file: $row.name name: ($row.name | path parse | get stem)}}
+        | upsert test {|module|
+            nu -c $'use ($module.file) *; $nu.scope.commands | select name module_name | to nuon'
+            | from nuon
+            | where module_name == $module.name
+            | where ($it.name | str starts-with "test_")
+            | get name
+        }
+        | flatten
+        | rename file module name
+    )
 
-        log info $"Run tests in ($module_name)"
-        let tests = (collect-commands $test_file $module_name $command)
+    if $list {
```

Comment:
Move this after the next paragraph. Consider having both `--list` and `--module` provided if I want to list the commands of a module.

Reply:
ooh i see :+1: